packages/token/src/tests/erc20/test_erc20.cairo

```diff
@@ -449,3 +460,159 @@ fn test_burn_from_zero() {
+    state.update(OWNER(), RECIPIENT(), VALUE);
+    assert_state_after_transfer(OWNER(), RECIPIENT(), VALUE);
+
+    spy.assert_only_event_transfer(contract_address, OWNER(), RECIPIENT(), VALUE);
+}
+
+#[test]
+#[should_panic(expected: ('ERC20: insufficient balance',))]
+fn test_update_from_non_zero_to_non_zero_insufficient_balance() {
+    let mut state = setup();
+    let contract_address = test_address();
+    let overflow_amt = SUPPLY + 1;
+
+    start_cheat_caller_address(contract_address, OWNER());
+    state.update(OWNER(), RECIPIENT(), overflow_amt);
+}
+
+#[test]
+fn test_update_from_non_zero_to_zero() {
+    let mut state = setup();
+
+    let mut spy = spy_events();
+    let contract_address = test_address();
+
+    start_cheat_caller_address(contract_address, OWNER());
+
+    assert_state_before_burn(OWNER());
+    state.update(OWNER(), ZERO(), VALUE);
+    assert_state_after_burn(OWNER(), VALUE);
+
+    spy.assert_only_event_transfer(contract_address, OWNER(), ZERO(), VALUE);
+}
+
+#[test]
+#[should_panic(expected: ('ERC20: insufficient balance',))]
+fn test_update_from_non_zero_to_zero_insufficient_balance() {
+    let mut state = setup();
+    let contract_address = test_address();
+    let overflow_amt = SUPPLY + 1;
+
+    start_cheat_caller_address(contract_address, OWNER());
+    state.update(OWNER(), ZERO(), overflow_amt);
+}
+
+#[test]
+fn test_update_from_zero_to_non_zero() {
+    let mut state = setup();
+
+    let mut spy = spy_events();
+    let contract_address = test_address();
+
+    start_cheat_caller_address(contract_address, RECIPIENT());
+
+    assert_state_before_mint(RECIPIENT());
+    state.update(ZERO(), RECIPIENT(), VALUE);
+    assert_state_after_mint(RECIPIENT(), VALUE);
+
+    spy.assert_only_event_transfer(contract_address, ZERO(), RECIPIENT(), VALUE);
+}
+
+#[test]
+fn test_update_from_zero_to_zero() {
+    // The update function should not be called from zero to zero
+    // because this does nothing to the state.
+    let mut state = setup();
+
+    let mut spy = spy_events();
+    let contract_address = test_address();
+    let supply_before_update = state.total_supply();
+
+    start_cheat_caller_address(contract_address, RECIPIENT());
+
+    state.update(ZERO(), ZERO(), VALUE);
+
+    let supply_after_update = state.total_supply();
+    assert_eq!(supply_before_update, supply_after_update);
+
+    spy.assert_only_event_transfer(contract_address, ZERO(), ZERO(), VALUE);
+}
+
+//
+// Helpers
+//
+
+fn assert_state_before_transfer(sender: ContractAddress, recipient: ContractAddress) {
+    let state = COMPONENT_STATE();
+    let initial_supply = SUPPLY;
+    let current_supply = state.total_supply();
+
+    assert_eq!(initial_supply, current_supply, "Incorrect supply before transfer");
+    assert_eq!(state.balance_of(sender), SUPPLY, "Incorrect balance before transfer");
+    assert_eq!(state.balance_of(recipient), 0, "Incorrect balance before transfer");
```

Comment:
Minor one:
```suggestion
    assert_eq!(initial_supply, current_supply, "Incorrect supply before transfer");
    assert_eq!(state.balance_of(sender), SUPPLY, "Incorrect sender balance before transfer");
    assert_eq!(state.balance_of(recipient), 0, "Incorrect recipient balance before transfer");
```

Reply:
IMO this doesn't add much for the potential inconsistencies it might generate. The rule is to add messages when it very clearly adds clarity. I don't feel this is the case, and if it is not that clear, we should not add them.